src/dash/extensions/BaseURLExtensions.js

```diff
@@ -292,204 +134,103 @@ Dash.dependencies.BaseURLExtensions = function () {
 
-                // If we didn't know where the SIDX box was, we have to look for it.
-                // Iterate over the data checking out the boxes to find it.
-                if (info.searching) {
-                    info.bytesLoaded = info.range.end;
-                    findSIDX.call(self, request.response, info, representation, function (segments) {
-                        if (segments) {
-                            callback.call(self, segments, representation, type);
+                var extraBytes = info.bytesToLoad,
+                    loadedLength = request.response.byteLength;
+
+                needFailureReport = false;
+                info.bytesLoaded = info.range.end - info.range.start;
+                isoFile = self.boxParser.parse(request.response);
+                sidx = isoFile.getBox("sidx");
+
+                if (!sidx || !sidx.isComplete) {
+                    if (sidx) {
+                        info.range.start = sidx.offset || info.range.start;
+                        info.range.end = info.range.start + (sidx.size || extraBytes);
+                    } else if (loadedLength < info.bytesLoaded) {
+                        // if we have reached a search limit or if we have reached the end of the file we have to stop trying to find sidx
+                        callback.call(self, null, representation, type);
+                        return;
+                    } else {
+                        var lastBox = isoFile.getLastBox();
+
+                        if (lastBox && lastBox.size) {
+                            info.range.start = lastBox.offset + lastBox.size;
+                            info.range.end = info.range.start + extraBytes;
+                        } else {
+                            info.range.end += extraBytes;
                         }
-                    });
+                    }
+                    loadSegments.call(self, representation, type, info.range, info, callback);
                 } else {
-                    segments = parseSegments.call(self, request.response, info.url, info.range.start);
-                    callback.call(self, segments, representation, type);
+                    var ref = sidx.references,
+                        loadMultiSidx,
+                        segments;
+
+                    if (ref !== null && ref !== undefined && ref.length > 0) {
+                        loadMultiSidx = (ref[0].reference_type === 1);
+                    }
+
+                    if (loadMultiSidx) {
+                        self.log("Initiate multiple SIDX load.");
+                        info.range.end = info.range.start + sidx.size;
+
+                        var j, len, ss, se, r, segs = [],
+                            count = 0,
+                            offset = (sidx.offset || info.range.start) + sidx.size,
+                            tmpCallback = function(segments) {
+                                if (segments) {
```

Comment:
Can we rename segments here in the callback to not hide / override segments in the outside?  Inverting segs / segments is probably enough.